### ISSUE TITLE
Errorneous "Artifact or version not found"

### ISSUE THREAD

**Issue by sideeffffect** [badges/shields#2696]
Are you experiencing an issue with...

- [x] [shields.io](https://shields.io/#/)
- [ ] My own instance
- [ ] [gh-badges NPM package](https://www.npmjs.com/package/gh-badges)

:beetle: **Description**

Shields can't find any version of the artifact `ch.epfl.scala:sbt-scalafix_2.10_0.13`, but the artifact is clearly there:
```
https://oss.sonatype.org/content/repositories/snapshots/ch/epfl/scala/sbt-scalafix_2.12_1.0/
```

:link: **Link to the badge**

https://img.shields.io/nexus/s/https/oss.sonatype.org/ch.epfl.scala/sbt-scalafix_2.12_1.0.svg

(discovered at https://github.com/scalacenter/sbt-scalafix/pull/35)

**Comment by sideeffffect** [comment]
yes, I can confirm that the group id is `ch.epfl.scala` and artifact id is `sbt-scalafix_2.12_1.0`

see another working example if it helps you:
```
https://oss.sonatype.org/content/repositories/releases/org/scalaz/scalaz-zio_2.12/
https://img.shields.io/nexus/r/https/oss.sonatype.org/org.scalaz/scalaz-zio_2.12.svg
```

**Comment by calebcartwright** [comment]
Thanks. I needed to confirm as that particular error message you are seeing only happens when the Nexus API returns no results. I'll need to dig in to see what the Nexus API is returning for that G & A

**Comment by calebcartwright** [comment]
So the API results using the Nexus Search API (https://oss.sonatype.org/service/local/lucene/search?g=ch.epfl.scala&a=sbt-scalafix_2.12_1.0) is returning an empty response:

```json
{
  "totalCount": 0,
  "from": -1,
  "count": -1,
  "tooManyResults": false,
  "collapsed": false,
  "repoDetails": [],
  "data": [],
}
```

**Comment by calebcartwright** [comment]
Also when I look at the pom files for `sbt-scalafix_2.12_1.0`, they all list the artifactId as `sbt-scalafix`: 

* https://oss.sonatype.org/content/repositories/snapshots/ch/epfl/scala/sbt-scalafix_2.12_1.0/0.5.10-11-cb063077-SNAPSHOT/sbt-scalafix-0.5.10-11-cb063077-SNAPSHOT.pom
* https://oss.sonatype.org/content/repositories/snapshots/ch/epfl/scala/sbt-scalafix_2.12_1.0/0.5.10-14-6a3dd917-SNAPSHOT/sbt-scalafix-0.5.10-14-6a3dd917-SNAPSHOT.pom
* https://oss.sonatype.org/content/repositories/snapshots/ch/epfl/scala/sbt-scalafix_2.12_1.0/0.5.6/sbt-scalafix-0.5.6.pom

Whereas the artifactId for the other artifacts (where the Nexus API _is_ returning the required data, and thus Shields can build badges) has the artifactId in the pom files matching (see https://oss.sonatype.org/content/repositories/releases/org/scalaz/scalaz-zio_2.12/0.1.0-0812841/scalaz-zio_2.12-0.1.0-0812841.pom for example)

This discrepancy is my best guess right now as to why the Nexus APIs are not returning any information for your artifact. We're entirely reliant on the Nexus API to deliver the data we need, so I don't really see much we can do on our end. 

Can you double check the artifactId's that you are using?

**Comment by sideeffffect** [comment]
@calebcartwright thanks for looking into this!
@olafurpg do you have any idea why the `sbt-scalafix` artifact has the wrong `artifactId` (just `sbt-scalafix`) and not the correct `sbt-scalafix_2.12_1.0`?

**Comment by sideeffffect** [comment]
the fact that `sbt-scalafix` doesn't have the `_2.12_1.0` part in `artifactId` is typical for `sbt` plugins (according to @olafurpg)

in that case, I've reported the issue with Sonatype, because the artifact is with the given id is clearly there, only Lucene can't find it:
https://issues.sonatype.org/browse/OSSRH-45426

**Comment by calebcartwright** [comment]
I've added the `needs-upstream-help` label based on the comment regarding invalid M2 repository layout in the [issue opened with Sonatype ](https://issues.sonatype.org/browse/OSSRH-45426?focusedCommentId=582150&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-582150)

@sideeffffect can you also add a link to the subsequent issue (referenced as NEXUS-18806) when you get a chance? I tried but was unable to locate it

**Comment by sideeffffect** [comment]
here's the link to the Nexus indexer issue: https://issues.sonatype.org/browse/NEXUS-18806

**Comment by okou19900722** [comment]
I also encountered a similar, but the message is `invalid response data`

https://img.shields.io/nexus/s/https/oss.sonatype.org/tk.okou/vertx-lang-jphp.svg

[![a](https://img.shields.io/nexus/s/https/oss.sonatype.org/tk.okou/vertx-lang-jphp.svg)](https://img.shields.io/nexus/s/https/oss.sonatype.org/tk.okou/vertx-lang-jphp.svg)

**Comment by calebcartwright** [comment]
Hi @okou19900722, would you mind opening a separate issue? These look like different issues even though they're both related to the Nexus service and I'd like to track them separately accordingly. 

Thanks!

**Comment by okou19900722** [comment]
ok, #2729

**Comment by calebcartwright** [comment]
Thanks @sideeffffect. I'm hitting an auth wall on that Nexus Indexer issue link so I've got no insight into if/how that's progressing. Please keep us posted!

**Comment by calebcartwright** [comment]
Since the underlying issue was that the indexer in Nexus 2 doesn't support invalid M2 repository layouts, I'm thinking (hoping 🤞) that this won't be a very common scenario.

Could we keep an eye on this for repeats, and if we see it again add Nexus version badge documentation along the lines of `This only works with valid M2 repository layouts for Nexus 2 targets`?

**Comment by paulmelnikow** [comment]
I see, it's a pretty specific issue.

**Comment by calebcartwright** [comment]
I sure hope so 😄
